rotkehlchen/data/import_utils.py

```diff
@@ -0,0 +1,283 @@
+                fee=trade_dict['fee_amount'],
+                fee_currency=trade_dict['fee_asset'],
+                link='',
+            ))
+        return trades
+
+    @classmethod
+    def process_entries(cls, timestamp: Timestamp, data: List[BinanceCsvRow]) -> int:
+        trades = cls.process_trades(timestamp=timestamp, data=data)
+        cls.db.add_trades(trades)
+        return len(trades)
+
+
+class BinanceStakingEntry(BinanceMultipleEntry):
+    @classmethod
+    def are_entries(cls, requested_operations: List) -> bool:
+        return (
+            requested_operations == ['ETH 2.0 Staking', 'ETH 2.0 Staking']
+        )
+
+    @classmethod
+    def process_entries(cls, timestamp: Timestamp, data: List[BinanceCsvRow]) -> int:
+        amount = abs(data[0]['Change'])
+        asset = data[0]['Coin']
+
+        ledger_action = LedgerAction(
+            identifier=0,
+            timestamp=timestamp,
+            action_type=LedgerActionType.EXPENSE,
+            location=Location.BINANCE,
+            amount=amount,
+            asset=asset,
+            rate=None,
+            rate_asset=None,
+            link=None,
+            notes='ETH 2.0 Staking',
+        )
+        db_ledger = DBLedgerActions(cls.db, cls.db.msg_aggregator)
+        db_ledger.add_ledger_action(ledger_action)
+        return 1
+
+
+class BinanceDepositWithdrawEntry(BinanceSingleEntry):
+    available_operations = ['Deposit', 'Withdraw']
+
+    @classmethod
+    def process_entry(cls, timestamp: Timestamp, data: BinanceCsvRow) -> None:
+        amount = data['Change']
+        asset = data['Coin']
+        category = AssetMovementCategory.DEPOSIT if data['Operation'] == 'Deposit' else AssetMovementCategory.WITHDRAWAL  # noqa: E501
+        if category == AssetMovementCategory.WITHDRAWAL:
+            amount = -amount
+
+        asset_movement = AssetMovement(
+            location=Location.BINANCE,
+            category=category,
+            address=None,
+            transaction_id=None,
+            timestamp=timestamp,
+            asset=asset,
+            amount=AssetAmount(amount),
+            fee=Fee(ZERO),
+            fee_asset=A_USD,
+            link='',
+        )
+        cls.db.add_asset_movements([asset_movement])
+
+
+class BinanceStakingRewardsEntry(BinanceSingleEntry):
+    available_operations = ['ETH 2.0 Staking Rewards', 'Launchpool Interest']
+
+    @classmethod
+    def process_entry(cls, timestamp: Timestamp, data: BinanceCsvRow) -> None:
+        asset = data['Coin']
+        amount = data['Change']
+        ledger_action = LedgerAction(
+            identifier=0,
+            timestamp=timestamp,
+            action_type=LedgerActionType.INCOME,
+            location=Location.BINANCE,
+            amount=amount,
+            asset=asset,
+            rate=None,
+            rate_asset=None,
+            link=None,
+            notes=data['Operation'],
+        )
+        db_ledger = DBLedgerActions(cls.db, cls.db.msg_aggregator)
```

Comment:
Could we avoid creating this objects over and over if you used a class instead of a namedtuple and have this as property? Also this would allow you to use abstracclass instead of raise the nonimplementederrors and improve error catching

Reply:
Yeah, sure, it appeared there during the development experiments, now I've removed them.
But could you please explain a bit your suggestion about error raising and catching, how would you do it?

Reply:
You can use ABCMeta like in https://github.com/rotki/rotki/blob/develop/rotkehlchen/interfaces.py#L8 The code won't allow you to create classes that don't implement correctly the methods and also the linter should warn you

Reply:
Ah I forgot that those were classmethods. Maybe is not possible to use both... I would need to check but at the moment I'm not sure